### ISSUE TITLE
Undefined index: tokens on line 227 module-elasticsuite-thesaurus/Model/Index.php

### ISSUE THREAD

**Issue by jsterling** [Smile-SA/elasticsuite#1165]
### Preconditions

Magento Version : CE 2.2.6

ElasticSuite Version : 2.6

Environment : Production

Third party modules :

### Steps to reproduce

1. perform a search (not selecting an option from autocomplete) 

### Expected result

1. Search results displayed

### Actual result

1. catalogsearch/result page throws a 500 error, log shows

`Notice: Undefined index: tokens in [webroot]/app\/vendor\/smile\/elasticsuite\/src\/module-elasticsuite-thesaurus\/Model\/Index.php on line 227"`

This is the line `foreach ($analysis['tokens'] as $token) {`

This happens for multiple users, from multiple different browsers including 'private browsing', however it only happens occasionally (maybe 1 in 10 searches). We're using Elastic Cloud - no errors or warnings shown.

**Comment by jsterling** [comment]
A very crude `if(!isset($analysis['tokens'])) $analysis['tokens'] = array();` just before line 227 seems to solve this

**Comment by romainruaud** [comment]
Well, seems kinda strange especially when I see the line just above this one : 

```php
        try {
            $analysis = $this->client->analyze(
                ['index' => $indexName, 'body' => ['text' => $queryText, 'analyzer' => $type]]
            );
        } catch (\Exception $e) {
            $analysis = ['tokens' => []];
        }
```

This would mean that we had a successful response of `$client->analyze` but without any tokens on it...

Are you able to reproduce this one easily ? I'm curious about what is actually being returned in the `$analysis` variable in the failing case.

**Comment by jsterling** [comment]
It was happening on virtually every search request until I added my change. We saw the same result whether or not there were Thesaurus entries.

I should be able to reproduce easily, if you can let me know the best format / way to send you a log or print and what you'd want to see?

**Comment by romainruaud** [comment]
I received your log, but I'm more interested by a `print_r($analysis)` just before the line which fails, to see what have been answered by the engine for the precedent call.

In any cases, PR #1176 will fix the issue and is scheduled to be released on monday with the new minor version.

**Comment by romainruaud** [comment]
Fixed by PR #1176 that will be part of next minor release.

I close
